### PR TITLE
Make admin-console deployable

### PR DESCRIPTION
**Pull request by sethvargo**
/assign @mikehelmick 

Part of https://github.com/google/exposure-notifications-server/issues/1273

**Release Note**

```release-note
Make admin-console deployable
```

### CODE REVIEW

**Review by sethvargo — COMMENTED** [review]
The diff is large, but it's mostly copy-paste. I had to rename some things to prevent a circular dependency though.

File: builders/server.dockerfile
Comment (mikehelmick) [inline builders/server.dockerfile:1]: why doesn't he existing service.dockerfile work?

**Review by mikehelmick — APPROVED** [review]
/hold
for the one question

File: builders/server.dockerfile
Comment (sethvargo) [reply builders/server.dockerfile:1]: Have to copy the assets over

Comment (sethvargo) [reply builders/server.dockerfile:1]: (templates)

File: cmd/admin-console/main.go
Comment (odeke-em) [inline cmd/admin-console/main.go:32]: If invoking done again is harmless, I'd encourage adding `defer done()` here so that regardless of what panics happen down below, it'll always be run, despite `done()` being explicitly invoked after `realMain(ctx)`

File: internal/admin/handle_authorized_apps.go
Comment (odeke-em) [inline internal/admin/handle_authorized_apps.go:46]: Minor nit, it took reading through a bunch of code blocks to see that the only permitted actions were "save" and "delete". Perhaps we could restructure this into a switch so
```go
switch form.Action {
default:
     ErrorPage(c, fmt.Sprintf("Invalid request, app to edit not found, with action: %q", form.Action))

case "delete":
      ...

case "save":
      ...
}
```

File: internal/admin/handle_authorized_apps.go
Comment (odeke-em) [inline internal/admin/handle_authorized_apps.go:196]: Minor nit on inversion order to reduce nesting in the more critical case
```go
if f.FormKey == "" {
     return ""
}

bs, err := base64...
...
return string(bytes)
```

File: internal/admin/handle_export_importers.go
Comment (odeke-em) [inline internal/admin/handle_export_importers.go:101]: This code seems duplicated also in `(*Server).HandleExportImportersSave` perhaps this helper
```go
func modelFromContext(c *gin.Context, db *model.ExportImportDB) (*model.ExportImport, erorr) {
    idRaw := strings.TrimSpace(c.Param("id"))
    if idRaw == "" || idRaw == "0" {
         // Just return a blank model.
         return new(model.ExportImport), nil
    }
   
    id, err := strconv.ParseInt(idRaw, 10, 64)
    if err != nil {
         return nil, fmt.Errorf("Failed to parse `id` param: %w", err)
    }

    model, err := db.GetConfig(ctx, id)
    if err != nil {
        err = fmt.Errorf("Failed to load export importer config: %w", err)
    }
    return model, err
}
```

File: internal/admin/handle_export_importers.go
Comment (odeke-em) [inline internal/admin/handle_export_importers.go:154]: Please beware that if at any point a time.Time value was created by unmarshaling Seconds and Nanosecond respectively and it got the zero value, these checks will fail as per https://play.golang.org/p/ma-1o9J4ynh or inlined  below
```go
package main

import (
	"encoding/json"
	"time"
)

func main() {
	type recv struct {
		Sec int64 `json:"sec"`
		Ns  int64 `json:"ns"`
	}
	rc := new(recv)
	if err := json.Unmarshal([]byte(`{}`), rc); err != nil {
		panic(err)
	}
	t := time.Unix(rc.Sec, rc.Ns)
	println("isZero", t.IsZero(), "unix <= 0", t.UnixNano() <= 0)
}
```
which prints `isZero false unix <= 0 true`

For that I'd highly recommend using `(time.Time).UnixNano() > 0`

File: internal/admin/handle_exports.go
Comment (odeke-em) [inline internal/admin/handle_exports.go:47]: If it is `""` this code will trip, but in prior checks you explicitly checked for `idParam != "" && idParam != "0"`. This is more impetus to use the refactored helper that I suggested above :-)

File: internal/admin/handle_exports.go
Comment (odeke-em) [inline internal/admin/handle_exports.go:100]: Ditto about `idParam == ""` too?

File: internal/admin/handle_health_authority.go
Comment (odeke-em) [inline internal/admin/handle_health_authority.go:86]: Does `idParam == ""` matter here?

File: internal/admin/handle_health_authority.go
Comment (odeke-em) [inline internal/admin/handle_health_authority.go:208]: Did you intend to capitalize it to "Thru-date", or did you mean "thru-date"?

File: internal/admin/handle_health_authority.go
Comment (odeke-em) [inline internal/admin/handle_health_authority.go:239]: Please simplify this check to
```go
return err
```

**Review by odeke-em — COMMENTED** [review]
Thank you for this change @sethvargo, and happy New Year to you and to the team! Just some comments, please take a look.

File: internal/admin/handle_authorized_apps.go
Comment (sethvargo) [reply internal/admin/handle_authorized_apps.go:46]: This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_authorized_apps.go
Comment (sethvargo) [reply internal/admin/handle_authorized_apps.go:196]: This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_export_importers.go
Comment (sethvargo) [reply internal/admin/handle_export_importers.go:101]: This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_export_importers.go
Comment (sethvargo) [reply internal/admin/handle_export_importers.go:154]: This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_exports.go
Comment (sethvargo) [reply internal/admin/handle_exports.go:47]: It's required to be a value by the router to reach this function. The router enforces a digit in the path.

This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_exports.go
Comment (sethvargo) [reply internal/admin/handle_exports.go:100]: It's required to be a value by the router to reach this function. The router enforces a digit in the path.

This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_health_authority.go
Comment (sethvargo) [reply internal/admin/handle_health_authority.go:86]: It's required to be a value by the router to reach this function. The router enforces a digit in the path.

This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_health_authority.go
Comment (sethvargo) [reply internal/admin/handle_health_authority.go:239]: This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

File: internal/admin/handle_health_authority.go
Comment (sethvargo) [reply internal/admin/handle_health_authority.go:208]: This is most likely a bug, but again, this PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.

**Comment by sethvargo** [comment]
Hi @odeke-em 

Thank you for the review. The items you identified are important and we'll be sure to address them in a follow-up PR. However, all these bugs already exist (this is a copy-paste and package rename of existing code), and I'd prefer to keep this PR isolated to _just_ the move. I can create another PR after this is merged fixing up the items you mentioned. I just don't want to couple a rename with changes in case we need to do a rollback.

**Pull request merged**
